kpi/deployment_backends/kc_access/shadow_models.py

```diff
@@ -130,20 +130,15 @@ def sync(cls, user):
         but updates `KobocatDigestPartial` in the KoBoCAT database instead of
         `PartialDigest` in the KPI database
         """
-        # Because of circular imports, we cannot decorate the method with
-        # `@kc_transaction_atomic`
-        from .utils import kc_transaction_atomic
-
-        with kc_transaction_atomic():
-            cls.objects.filter(user=user).delete()
-            # Query for `user_id` since user PKs are synchronized
-            for partial_digest in PartialDigest.objects.filter(user_id=user.pk):
-                cls.objects.create(
-                    user=user,
-                    login=partial_digest.login,
-                    confirmed=partial_digest.confirmed,
-                    partial_digest=partial_digest.partial_digest,
-                )
+        cls.objects.filter(user=user).delete()
+        # Query for `user_id` since user PKs are synchronized
+        for partial_digest in PartialDigest.objects.filter(user_id=user.pk):
+            cls.objects.create(
+                user=user,
+                login=partial_digest.login,
+                confirmed=partial_digest.confirmed,
+                partial_digest=partial_digest.partial_digest,
+            )
```

Comment:
Because `KobocatDigestPartial.sync()` is only used in `KobocatUser.sync()` which is called only inside the `post_save` signal of the `User` model. 
So I just [moved it](https://github.com/kobotoolbox/kpi/pull/4558/files#diff-cbf9041792243e92abae8e95f9af2c66be3e10b1f67292c01bf2d1520d6cf331R49-R52) to wrap all the kobocat queries inside the same transaction. 

I can leave a comment which explains all of that. 
What do you think?

Reply:
Added the comment.